scapy/utils.py

```diff
@@ -1500,7 +1500,14 @@ def _read_packet(self, size=MTU):
         if len(hdr) < 16:
             raise EOFError
         sec, usec, caplen, wirelen = struct.unpack(self.endian + "IIII", hdr)
-        return (self.f.read(caplen)[:size],
+
+        try:
+            data = self.f.read(caplen)[:size]
+        except OverflowError as e:
+            warning(f"Pcap: {e}")
```

Comment:
I have to admit I haven't figured out where the overflow comes from yet but it seems that that issue is probably unrelated to the OSS-Fuzz crash. I think the test itself can be "fixed" by replacing the OSS-Fuzz testcase with something like
```sh
b'\xd4\xc3\xb2\xa1\x02\x00\x04\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xff\x00\x00\x01\x00\x00\x00\x04{\xdcf\xc2\xa5\x07\x008\x00\x00\x008\x00\x00\x00A]+\xdb]\x04\x8e(6\n\x99\xcb\x08\x00E\x00\x00*\x00\x01\x00\x00@\x06\xe3V\x07\x87\xa5m\x17\x15\xd3m\x01\x85\x01\x85\x00\x00\x00\x00\x00\x00\x00\x00P\x02 \x00\xc5_\x00\x000\x00'
```
it triggers the crash without 1935723c and doesn't trigger the overflow. The overflow can probably be looked into separately with a separate testcase without blocking the next release (as far as I understand it's coming) and distros running the test suite on those platforms.

Reply:
(I opened https://github.com/secdev/scapy/pull/4530 with the test case)